benches/themis/benches/secure_message_encrypt_decrypt_ecdsa.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2020 Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+
+use libthemis_sys::{
+    themis_secure_message_decrypt, themis_secure_message_encrypt, THEMIS_BUFFER_TOO_SMALL,
+    THEMIS_SUCCESS,
+};
+use themis::{keygen::gen_ec_key_pair, secure_message::SecureMessage};
+
+const KB: usize = 1024;
+const MB: usize = 1024 * KB;
+#[allow(clippy::identity_op)]
+const MESSAGE_SIZES: &[usize] = &[
+    16,     // UUID
+    32,     // SymmetricKey
+    64,     // cache line (and close to EcdsaPrivateKey)
+    256,    // RsaPrivateKey
+    4 * KB, // memory page
+    16 * KB,
+    64 * KB,
+    1 * MB, // L2 cache
+    2 * MB,
+    4 * MB,
+];
+
+pub fn encryption(c: &mut Criterion) {
+    let (private, public) = gen_ec_key_pair().split();
+    let (private, public) = (private.as_ref(), public.as_ref());
+
+    // Allocate buffer large enough for maximum test
+    let mut encrypted = vec![0; 10 * MB];
```

Comment:
FYI: `b.iter()` it the hot loop in benchmarks. It's okay to allocate and reallocate outside of that loop. You can, for example, do something like
```rust
let message = vec![0; size];
let mut encrypted = vec![0; size + 1024];
```
The current approach is fine too. (But you do allocate `message` every time.)